test/framework/clustervalidator.go

```diff
@@ -81,14 +81,18 @@ func buildClusterClient(kubeconfigFileName string) (client.Client, error) {
 }
 
 func buildClusterSpec(ctx context.Context, client client.Client, config *cluster.Config) (*cluster.Spec, error) {
-	clus := &v1alpha1.Cluster{}
-	key := machinerytypes.NamespacedName{Namespace: config.Cluster.Namespace, Name: config.Cluster.Name}
-	if err := client.Get(ctx, key, clus); err != nil {
-		return nil, fmt.Errorf("failed to get cluster to build spec: %s", err)
+	clusterConfig := config.DeepCopy()
+	// The cluster config built by the test does not have certain defaults like the bundle reference,
+	// so fetch that information from the cluster if missing. This is needed inorder to build the cluster spec.
+	if clusterConfig.Cluster.Spec.BundlesRef == nil {
```

Comment:
How do we validate "default" bundle then, like the case that only update mgmt bundle, how do we make sure wl cluster bundle doesn't get effected?

Reply:
So, in that test we just grab the EKSA Cluster object from the cluster before upgrading the management cluster and read the bundleRef from there. After updating the management cluster, we grab it again for the updated bundle.

The CLI has some logic that "defaults" the bundle based on different version details and then applies it to the cluster. It's kind of complex to replicate so this code here is to sort bridge that gap in the test suite (I think the controller does something similar as well)